### PR TITLE
Main nav and side nav updates for developer portal IA updates 1153

### PR DESCRIPTION
**Pull request by mjlumetta**
## Updates to the site main and side navigation for the IA changes

Ticket: https://github.com/department-of-veterans-affairs/vets-contrib/issues/1153 

This ticket changes the main navigation under the header to include only the Documentation and Support links and updates the styling of the main nav and side nav to match the new [desktop](https://www.figma.com/file/SGQZyWGE44uYM4lI1twVsw/IA-Updates-(Jan-2019)?node-id=0%3A1) and [mobile](https://www.figma.com/file/SGQZyWGE44uYM4lI1twVsw/IA-Updates-(Jan-2019)?node-id=85%3A131) designs. The main nav is located under the header on desktop and opened by clicking the Menu button on mobile.

### Testing

Check that the design and behavior matches the mockups for [desktop](https://www.figma.com/file/SGQZyWGE44uYM4lI1twVsw/IA-Updates-(Jan-2019)?node-id=0%3A1) and [mobile](https://www.figma.com/file/SGQZyWGE44uYM4lI1twVsw/IA-Updates-(Jan-2019)?node-id=85%3A131).

#### Desktop

* Check that if you are on a documentation page (path starting with `/explore`), the Documentation nav link in the main link is highlighted with the yellow bottom border and text. If you are not on a documentation page, this link should not be highlighted. (Desktop - 1 Nav)
* Check that if you hover over either Documentation or Support on any page, the nav link is highlighted and the other is not. Test both on a documentation page and a non-documentation page. (Desktop - 1a Nav hover example)
* Check that any nav link is highlighted with an all-sides yellow border when active/clicked. (Desktop - 1b Nav press example)
* Check that the side nav matches on documentation pages. On the main documentation page (`/explore`), Overview should be selected. On the specific API pages - Benefits, Facilities, Health, and Veteran Verification - the corresponding list item should be selected and its sublist should be expanded. (Desktop - 2 Sidenav, 2a Sidenav selected)
* On any page, hovering over either a main or secondary sidenav list item should cause it to have a gray background. (Desktop - 2 Sidenav, 2b Sidenav secondary hover)
* When a secondary sidenav item is selected, it should be bolded. (Desktop - 2c Sidenav secondary selected)

#### Mobile

* Check that the main navigation includes a working close button in the top right, the Get Started button which links to `/apply`, and the two nav links for Documentation and Support.
* The Documentation nav link should have a + icon on its far right, and clicking it should toggle open a submenu with Overview, Benefits, Facilities, Health, and Veteran Verification (switching the + to a - icon). Each of the submenu links should be a link, and the Documentation nav link (technically a button) should not link to anything. 
* The sub nav links for Documentation should be bolded when on that page. 
* The Support nav link should have no icon and clicking it should open an email to api@va.gov.
* On click/press, the main nav links and secondary nav links should be surrounded by a solid yellow border.

#### Both

* Check that the header is left-aligned and says "VA | Developer Portal".
* Check that the Support nav link opens an email to api@va.gov.
* Test intermediate browser sizes.

### Questions

Unsure what automated tests to write for this (either unit or e2e - assume unit would be more applicable for these changes). If anyone has suggestions about what tests to add around these components, I'd appreciate them.

### CODE REVIEW

File: src/components/NavBar.tsx [hunk 1]
```diff
@@ -36,66 +53,110 @@ export class NavBar extends React.Component<INavBarProps, INavBarState> {
...
-              <li>
-                <Link to ="/explore/health" className="usa-nav-link">Health</Link>
-              </li>
-              <li>
-                <Link to ="/oauth" className="usa-nav-link">OAuth</Link>
-              </li>
-              <li>
-                <Link to ="/explore/verification" className="usa-nav-link">Verification</Link>
+              <li className="main-nav-item">
+                <a href="mailto:api@va.gov" target="_blank" className="usa-nav-link"
```
Comment (jeff2d2) [inline src/components/NavBar.tsx]:
```suggestion
                <a href="mailto:api@va.gov" className="usa-nav-link"
```
The `target="_blank"` attribute here is going to cause a new tab to be opened before calling the default mail client. Removing this will fix that.

**Comment by jeff2d2** [comment]
I see the new `Overview` left nav item as being highlighted when I click on `Documentation`. As soon as I start scrolling, the left nav jumps to `Benefits` right away. Just curious if this is expected behavior? Is `Overview` just an empty section that we don't yet have copy for?

File: src/components/NavBar.tsx [hunk 2]
```diff
@@ -1,26 +1,43 @@
+import classNames from 'classnames';
```
Comment (juliaelman) [inline src/components/NavBar.tsx]: Needed to manually add locally. Might need to be declared in `package.json`.

File: src/components/NavBar.tsx [hunk 3]
```diff
@@ -1,26 +1,43 @@
...
 import * as React from 'react';
-import { Link } from 'react-router-dom';
+import MediaQuery from 'react-responsive';
```
Comment (juliaelman) [inline src/components/NavBar.tsx]: Why was this added? What is the benefit of adding?

**Comment by jeff2d2** [comment]
`"@types/classnames": "^2.2.7"` is missing as a package dependency.

Exact error:
```
Failed to compile
/Users/jdunn/Documents/GitHub/developer-portal/src/components/NavBar.tsx
ERROR in /Users/jdunn/Documents/GitHub/developer-portal/src/components/NavBar.tsx(1,24):
TS7016: Could not find a declaration file for module 'classnames'. '/Users/jdunn/Documents/GitHub/developer-portal/node_modules/classnames/index.js' implicitly has an 'any' type.
  Try `npm install @types/classnames` if it exists or add a new declaration (.d.ts) file containing `declare module 'classnames';`
This error occurred during the build time and cannot be dismissed.
```

File: src/components/NavBar.tsx (hunk 3 above)
Comment (dvogel) [reply src/components/NavBar.tsx]: @mjlumetta gave a nice summary of where this fits in on slack: https://lighthouseva.slack.com/archives/CAZF880HW/p1550782991000400

File: src/containers/ExploreDocs.tsx [hunk 4]
```diff
@@ -74,16 +70,20 @@ function SideNavCategoryEntry(currentUrl: string, apiCategoryKey: string, apiCat
     );
 }
 
-function SideNav({ match: { url } } : RouteComponentProps<IApiNameParam>) {
-  let idx = 0;
+export function SideNav({ match: { url } } : RouteComponentProps<IApiNameParam>) {
   const buildSideNavCategoryEntry = (key: string) => {
-    return SideNavCategoryEntry(url, key, apiDefs[key], idx++);
+    return SideNavCategoryEntry(url, key, apiDefs[key]);
```
Comment (dvogel) [inline src/containers/ExploreDocs.tsx]: Since the purpose of this closure is to automatically increment `idx` with each call, you should be able to inline the contents of the closure where it is called.

File: src/types/constants.ts [hunk 5]
```diff
@@ -44,3 +44,20 @@ export type SUBMIT_APPLICATION_ERROR = typeof SUBMIT_APPLICATION_ERROR;
 
 export const TOGGLE_ACCEPT_TOS = 'TOGGLE_ACCEPT_TOS';
 export type TOGGLE_ACCEPT_TOS = typeof TOGGLE_ACCEPT_TOS;
+
+// Media query values - see src/base.scss and node_modules/uswds/src/stylesheets/core/_variables.scss
+export const MEDIUM_SCREEN = '640px';
+export const MEDIUM_SCREEN_QUERY = 'screen and (min-width: 640px)';
+export const NOT_MEDIUM_SCREEN_QUERY = 'screen and (max-width: 639px)';
```
Comment (dvogel) [inline src/types/constants.ts]: How about `SMALLER_` and `LARGER_` instead of `NOT_MEDIUM_` and `MEDIUM_`? That way I don't have to check the definition of either constant to get an understanding of when each will apply.

File: src/components/NavBar.tsx (hunk 1 above)
Comment (mjlumetta) [reply src/components/NavBar.tsx]: is that the pattern we normally follow? if so, removing it

Comment (jeff2d2) [reply src/components/NavBar.tsx]: Sorry, @mjlumetta I'm not familiar with if that's the pattern or not. I was just noting the issue and what a possible fix might be. For other sites though, I don't think that clicking an email link normally opens a new blank tab and then launches the default mail client.

Comment (juliaelman) [reply src/components/NavBar.tsx]: Agree with @jeff2d2. Let's remove.

File: src/types/constants.ts (hunk 5 above)
Comment (mjlumetta) [reply src/types/constants.ts]: switching to `OVER_` and `UNDER_`. or i could see an argument for not having anything other than the size values and leaving query construction up to authors of individual components

File: src/components/NavBar.tsx (hunk 1 above)
Comment (mjlumetta) [reply src/components/NavBar.tsx]: ohhh i see, sorry - i have mine set up to open in gmail, so it doesn't look like an issue for me. yeah, i see the benefit there.

**Comment by mjlumetta** [comment]
@jeff2d2 the scrolling issue you noted is a good question that i'm not sure i can answer on my own. there's some kind of functionality that i haven't quite looked into with the scrolling so that when you scroll through the documentation, it automatically updates the dynamic routes for each API section. as a result, you can't get back to Documentation/Overview by scrolling because Benefits is usually still in the window. i'm not sure what to do about that or if removing this behavior is within the scope of this ticket (had thought that it was not). curious what others think about solutions or if anyone knows the ticket where that work is supposed to be done, this one or another one. @dfaithberry @dvogel @juliaelman 

Julia, vis a vis the other changes to the banner/VCL, those are implemented on a separate branch. i had actually done most of the work with those changes merged in, but i pulled them apart to prevent git conflicts and to be clear about what was done where. i think that one should go first, so when they're together, it should look correct. is that sufficient? either way, i should pull/rebase off master before merging, so i can check the content there.

**Comment by juliaelman** [comment]
:shipit:
